api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -156,7 +161,48 @@ private void processOverloadedRuns(final List<PipelineRun> runs) {
                 .filter(pod -> isPodUnderPressure(pod.getValue(), thresholds))
                 .collect(Collectors.toList());
 
+        final List<PipelineRun> runsToUpdateTags = getRunsToUpdatePressuredTags(running, runsToNotify);
         notificationManager.notifyHighResourceConsumingRuns(runsToNotify, NotificationType.HIGH_CONSUMED_RESOURCES);
+        return runsToUpdateTags;
+    }
+
+    private List<PipelineRun> getRunsToUpdatePressuredTags(final Map<String, PipelineRun> running,
+                              final List<Pair<PipelineRun, Map<ELKUsageMetric, Double>>> runsToNotify) {
+        final List<PipelineRun> listOfRunsToNotify = runsToNotify
+                .parallelStream()
+                .map(Pair::getLeft)
+                .collect(Collectors.toList());
+        final Predicate<PipelineRun> runToBeNotified = listOfRunsToNotify::contains;
+        final Stream<PipelineRun> runsToAddTag = listOfRunsToNotify
+                .parallelStream()
+                .filter(this::hasNoPressuredTag)
+                .map(this::addPressuredTagToRun);
+        final Stream<PipelineRun> runsToRemoveTag = running.values()
+                .parallelStream()
+                .filter(runToBeNotified.negate())
+                .filter(this::hasPressuredTag)
+                .map(this::removePressuredTagFromRun);
+        return Stream.concat(runsToAddTag, runsToRemoveTag).collect(Collectors.toList());
+    }
+
+    private boolean hasPressuredTag(final PipelineRun run) {
```

Comment:
try to get rid of such small methods, I believe that in stream context it will obvious without methods. f.e.:

final Stream<PipelineRun> runsToRemoveTag = running.values()
          .parallelStream()
          **.filter(run -> listOfRunsToNotify.contains(run))
          .filter(run -> run.getTags().contains(UTILIZATION_LEVEL_HIGH))**

Reply:
also lets think that if `run.getTags().contains(UTILIZATION_LEVEL_HIGH)` the value is `true` because we don't have other option